assets/js/theme/global/modal.js

```diff
@@ -196,12 +210,65 @@ export class Modal {
         this.$content.html('');
     }
 
+    setupFocusableElements(modalType) {
+        this.$lastPageFocusedElement = $(document.activeElement);
+
+        const $collection = focusableElements[modalType]();
+        $collection.each((index, element) => $(element).removeClass(focusedClass)); 
+        const $elementToFocus = $($collection.get(0));
+        $elementToFocus.focus().addClass(focusedClass);
+
+        $('#modal').on('keydown', this.onTabbing.bind(null, modalType));
+    }
+
+    onTabbing(modalType, event) {
+        const isTAB = event.which === TAB;
+
+        if (!isTAB) return;
+
+        const $collection = focusableElements[modalType]();
+        const collectionLastIdx = $collection.length - 1
+
+        const $currentElement =  $collection.filter(`.${focusedClass}`);
+        const currentElementIdx = $collection.index($currentElement);
+
+        const direction = event.which === TAB && event.shiftKey ? 'backwards' : 'forwards';
+
+        let nextElementIdx;
+        let $candidatesCollection;
+        if (direction === 'forwards') {
+            nextElementIdx = currentElementIdx === collectionLastIdx
+                ? 0
+                : currentElementIdx + 1;
+            $candidatesCollection = $collection.slice(nextElementIdx);
+        } else if (direction === 'backwards') {
+            nextElementIdx = currentElementIdx === 0
+                ? collectionLastIdx
+                : currentElementIdx - 1;
+            $candidatesCollection = $($collection.slice(0, nextElementIdx + 1).get().reverse());
+        }
+
+        $candidatesCollection.each((index, element) => {
+            const $focusCanditate = $(element);
+            $focusCanditate.focus();
```

Comment:
Why do we need to trigger focus event handler on every collection element?

Reply:
Because not all elements in focusable elements collection truly can be focused - because of their own reasons. For example in video I added we have wishlist form. Inputs in it should be focusable, but they are not. And when you trying to set focus to first of this input - focus wont affect to this element and will stay on previous. But in such case we need to focus on next focusable element. So I am trying to find first truly focusable element in collection

Reply:
I think in this case we need to change the logic of collection creation, we need create a collection of truly focusable elements

Reply:
Maybe. But in that new logic we will need to check all elements parents on aria-hidden, tabindex=-1 and maybe other attributes to detect they are not focusable. It can be even more painful for performance and readability

Reply:
In result of call with Yury, we have made a decision to proceed with current algorithm